test/relations.test.js

```diff
@@ -1947,30 +2671,68 @@ describe('relations', function () {
           supplierId = supplier.id;
           should.not.exist(e);
           should.exist(supplier);
-          supplier.account.create({accountNo: 'a01'}, function (err, account) {
+          supplier.account.create({accountNo: 'a01', block: false}, function (err, account) {
             supplier.account(function (e, act) {
               accountId = act.id;
               should.not.exist(e);
               should.exist(act);
               act.should.be.an.instanceOf(Account);
+              act.block.should.be.false;
               supplier.account().id.should.equal(act.id);
               act.supplierName.should.equal(supplier.name);
               done();
             });
           });
-        });
+        });
+      });
+    });
+
+    it('should find record that match scope', function (done) {
+      Account.updateAll({ block: true }, function (err) {
+        Supplier.findById(supplierId, function (err, supplier) {
+          supplier.account(function (err, account) {
+            should.not.exists(account);
+            done();
```

Comment:
@bajtos Not sure to which comment you are referring...was it the one about the test failing when `refresh=true`?
I mentioned elsewhere that I believe this test was broken, not the code being tested.
It passed before, when `refresh=false` (default case), because the account was cached, but the test didn't check for `act.block === false`, and indeed, the property was not set.